### PR TITLE
op-chain-ops: add cross domain utils

### PR DESCRIPTION
**Pull request by tynes**
**Description**

We need a list of pending withdrawals to be able to
migrate the withdrawal hashes to the new withdrawal
hash scheme. This commit implements the script for
fetching pending withdrawals as well as a unit test
for the function responsible for fetching pending withdrawals.

From within `op-chain-ops`, the script can be ran as follows:

```bash
$ go run cmd/withdrawals/main.go
```

### CODE REVIEW

File: op-chain-ops/crossdomain/encoding_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzVersionedNonce(f *testing.F) {
+	f.Fuzz(func(t *testing.T, _nonce []byte, _version uint16) {
+		inputNonce := new(big.Int).SetBytes(_nonce)
+
+		// Clamp nonce to uint240
+		if inputNonce.Cmp(crossdomain.NonceMask) > 0 {
+			fmt.Println("claming..")
```
Comment (tynes) [inline op-chain-ops/crossdomain/encoding_test.go:18]: delete this dead code

File: op-chain-ops/crossdomain/withdrawals.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+				return nil, fmt.Errorf("expected version %d, got version %d", version, msg.Version())
+			}
+		}
+
+		hash, err := msg.Hash()
+		if err != nil {
+			return nil, err
+		}
+
+		if err != nil {
```
Comment (optimisticben) [inline op-chain-ops/crossdomain/withdrawals.go:113]: Redundant error check

**Review by optimisticben — COMMENTED** [review]
lgtm, 1 small change

File: op-chain-ops/crossdomain/withdrawals.go (hunk 2 above)
Comment (tynes) [reply op-chain-ops/crossdomain/withdrawals.go]: Good catch, thank you!

**Review by optimisticben — APPROVED** [review]
lgtm

**Review by mslipper — APPROVED** [review]

**Pull request merged**